src/hotspot/share/runtime/semaphore.hpp

```diff
@@ -50,7 +50,12 @@ class Semaphore : public CHeapObj<mtSynchronizer> {
   Semaphore(uint value = 0) : _impl(value) {}
   ~Semaphore() {}
 
-  void signal(uint count = 1) { _impl.signal(count); }
+  void signal(uint count = 1) { _impl.signal(count, false /* ignore_overflow */); }
+
+  // Ignore error of overflow
+  void signal_overflow(uint count = 1) {
```

Comment:
I'm a bit unsure about how to name and describe this given that the overflow check only relates to debug builds - in a product build both `signal` and `signal_overflow` are identical.

I had missed previously that the `ignore_overflow` parameter was only in the impl API not the main public API.

Reply:
Thanks you for reviewing this change. 

The reason we only had ignore_overflow in impl API because I defined my own platform-independent Semaphore class, aka. AsyncLogSemaphore in previous revision.  I changed type of _sem from Sempahore to AsyncLogSemaphore. I "overloaded" signal(uint count) in that way so  I don't  pollute "public API".


> I don't think this is necessary, just call wait(1, true) where needed.

I think your opinion is that it's not necessary.  how about I just add a second parameter ignore_overflow to Semaphore::signal with the default value = false?